### PR TITLE
Settings page: UI for feature additions in existing addons

### LINKED ISSUE #1424: "Update" Tooltip
Update-scratchaddons-tooltip

I thought when a tool/modification gets updated recently it will show what's new in that tool/modification in a update toolotip.
Or
There also should be a update log on the tool's/modification's box.

This could allow users to pay more attention to the new feature in the tool/modification.

### PR DESCRIPTION
**Pull request by lisa-wolfgang**
Resolves #1424

### Changes to settings page

The settings page now looks like this:

image

Addons that are designated as updated now appear at the bottom of the "New" categories, which have been renamed accordingly to "Featured new addons and updates" and "Other new addons and updates". As a consequence, the tag sort order has been tweaked.

A description can be specified for the update, which appears at the top of the notices section. Addon settings can also be designated as updated and will have a New tag next to them.

image

Just like the "New" tag, these indicators will automatically disappear in subsequent releases.

### Changes to addon manifest

A new property has been added.

**`latestUpdate` (object, optional)**
Provides information to the settings page about the latest update to this addon. Responsible for the "New options" tag and the New tag on individual addon settings. If not specified, no update-specific tags will be added.

Sub-properties:
- `"version"` (string, required) The version that this update applies to. If the value is the same as the current version of the extension, the addon will get the "New options" tag and display the indicators (if any) specified in `temporaryNotice` and `newSettings`.
- `"isMajor"` (boolean, optional) Indicates whether the addon should be included in the "Featured new addons and updates" category. Use the same discretion that you would for the Featured tag on regular new addons. Defaults to `false` if not specified.
- `"temporaryNotice"` (string, optional) A text description of the new features or changes. This will appear as a notice-style banner at the top of the info section (above any warnings or notices) when the addon has the "New options" tag. If not specified, no banner will be displayed.
- `"newSettings"` (array of strings, optional) The IDs of any settings that should be designated as new in this update. These settings will display the New tag when the addon has the "New options" tag. If not specified, no settings will display the New tag.

Example:
```json
"latestUpdate": {
  "version": "1.22.0",
  "isMajor": true,
  "temporaryNotice": "You can now set a whole lotta new settings.",
  "newSettings": ["test-2", "test-3"]
},
```
If these changes are merged, the manifest schema will need to be updated.

### Reason for changes

Feature additions to addons are currently not showcased anywhere on the settings page. As some new features are particularly noteworthy, there should be a way to showcase them alongside the new addons.

### Tests

Tested on browser w/ Chromium 95 and Firefox 94.0.1.

### CODE REVIEW

**Comment by apple502j** [comment]
Is there a way to indicate an update without options?

**Comment by mxmou** [comment]
> Is there a way to indicate an update without options?

A better name for the tag might be "Updated" with a tooltip that describes what is updated, as suggested in #1424.

**Comment by mxmou** [comment]
Could updated addons be a separate group on the setting page?

**Comment by lisa-wolfgang** [comment]
> Is there a way to indicate an update without options?

Not at the moment. We would need to discuss the best UI for this.

This shouldn't be a tooltip because users wouldn't expect that kind of information to go there, so few people would hover over the tag. The latest update should be alongside the warnings and notices in the respective extension version, and in future versions, it should be automatically appended to the addon description. Of course, if we decide that this is the ideal strategy, we would need to implement something like #2384.

> A better name for the tag might be "Updated" with a tooltip that describes what is updated, as suggested in #1424.

I chose the name "New options" because it hints to the user that they need to open the addon settings to enable and/or learn about the features that the update adds. "Updated" doesn't even suggest that new functionality was added, leading people to think that it's not important enough to deserve their attention, which would defeat the purpose of this feature.

> Could updated addons be a separate group on the setting page?

I dislike this because it further increases the perceived importance gap between updates and new addons. Some addon updates are significantly more useful than some non-featured addons. That's why I added `isMajor` -- most updates will probably go under the "Other" category, but anything feature-worthy has the option to be displayed in the "Featured" category.

**Comment by lisa-wolfgang** [comment]
More ideas for UI:
- Having the update description display below the addon title

image

- Auto-expanding the addon (suggested by Biker2000: https://discord.com/channels/806602307750985799/809063831841407018/910008475076165632)

**Comment by lisa-wolfgang** [comment]
I think it might be better to place that update description notice at the top of the notices section inside the addon. Hiding the addon description would make it harder for users who haven't seen the addon to figure out what it does.

Automatically appending the update descriptions to the main addon description in future versions also seems like a bad idea in retrospect.

I don't think the lack of update description support should prevent this addon from being merged for v1.23 --  https://github.com/ScratchAddons/ScratchAddons/pull/2010, https://github.com/ScratchAddons/ScratchAddons/pull/3825, https://github.com/ScratchAddons/ScratchAddons/pull/3804, https://github.com/ScratchAddons/ScratchAddons/pull/3457, and https://github.com/ScratchAddons/ScratchAddons/pull/3710 are features that would greatly benefit from this, especially since customizable website colors have been very highly requested. (It would also help beef up the new addon section this release...)

**Comment by lisa-wolfgang** [comment]
Known issue: the update description banner has a slightly greater height than regular notices and warnings.

image

This is because the New tag that is used as an icon has a greater height than the icons for notices and warnings, and I don't think there's a simple way to fix this. (Besides possibly removing the New tag? Most update descriptions will be pretty obvious that they're talking about updates, and the purple color is the same as the New tag, although color isn't the most accessible indicator.)

**Comment by lisa-wolfgang** [comment]
Something else to discuss is how to decide how "important" an update has to be to go alongside new addons.

Obviously, new features and settings should be recognized as well as newly split addons.
Not so sure about bug fixes and performance enhancements, unless they're really significant. Same goes for presets and preset updates.
Definitely not things related to the presentation on the settings page such as modified tags, with the possible exception of when an addon moves out of beta.

**Comment by CST1229** [comment]
What about showing the update description instead of the addon description for _people who have the addon enabled_?

**Comment by lisa-wolfgang** [comment]
> What about showing the update description instead of the addon description for _people who have the addon enabled_?

I like the idea, but it might be a little inconsistent/confusing to have enabled addon updates and disabled addon updates side-by-side.

File: addons/test-to-remove/addon.json [hunk 1]
```diff
@@ -0,0 +1,81 @@
+{
+  "name": "Test to remove",
+  "description": "This is a test for the update label PR.",
+  "tags": ["editor", "beta"],
+  "info": [
+    {
+      "type": "warning",
+      "text": "This addon should be deleted before the PR is merged, and its entry in addons.json should also be removed.",
+      "id": "wl-dont-forget"
```
Comment (WorldLanguages) [inline addons/test-to-remove/addon.json:9]: LOL

**Comment by apple502j** [comment]
Manifest: ScratchAddons/manifest-schema#51

**Pull request merged**